:arrow_up: refactor code to work with despina

matthiasgeihs opened the pull request.

ggwpez commented on cmd/demo/node.go:
```diff
@@ -115,7 +113,11 @@ func (n *node) peer(addr perunpeer.Address) *peer {
 
 func (n *node) setupChannel(ch *client.Channel) {
-	perunID := ch.Peers()[0]
+	if len(ch.Peers()) > 2 {
```
I think this check does not make sense, since *go-perun* can only do two-party channels in *Despina*.  
The version of *go-perun* is fixed anyway, so there is no way this changes unless we update it. If you still want to have the check, you also need to do it in the `HandleProposal` function.

matthiasgeihs replied:
i vote for including both checks, when initiating a channel opening and when receiving a proposal. the reason is that a developer who uses our code should be aware that the code only works for the two-party case. this is why i added also comments on the respective lines of code. i also changed the action from warning to fail, because code execution should not continue running if for some reason we have other than 2 participants in the channel.

ggwpez replied:
Well, the code is not meant to be used by other developers since we do not have docs or tests (and it is just a demo). It also does not work for sub-channels, so another test to check `Locked` for zero-sum would be needed or a comment.

ggwpez commented on cmd/demo/node.go:
```diff
@@ -25,23 +25,25 @@ import (
 	"perun.network/go-perun/wallet"
+	"perun.network/go-perun/wire"
+	wirenet "perun.network/go-perun/wire/net"
+	wirenetsimple "perun.network/go-perun/wire/net/simple"
```
```suggestion
	simple "perun.network/go-perun/wire/net/simple"
```
We only use names like `ethchannel` sometimes because there are multiple `channel`. For the `simple` there is only one so it should be fine just naming it `simple`.

matthiasgeihs replied:
ok, the reasoning was that "simple" is a bit of a too simple name to be understood what this refers to when out of context. but i guess the context is given and, as you suggest, only using package aliases when there is a package name overload seems to be the standard way to do it.

ggwpez reviewed (changes requested):
Looks good

matthiasgeihs committed:
:arrow_up: refactor code to work with despina

- change dependency in go.mod
- refactor code to use wire bus and wire net
- fail on channel opening if not 2-party channel
- use dummy app def to avoid usage of wallet test package

ggwpez commented on cmd/demo/node.go:
```diff
@@ -115,7 +113,11 @@ func (n *node) peer(addr perunpeer.Address) *peer {
 func (n *node) setupChannel(ch *client.Channel) {
-	perunID := ch.Peers()[0]
+	if len(ch.Peers()) != 2 {
+		log.Fatal("Channels with more than 2 participants are currently not supported")
```
```suggestion
		log.Fatal("Only channels with two participants are currently supported")
```

matthiasgeihs replied:
done

ggwpez commented on cmd/demo/node.go:
```diff
@@ -187,6 +189,10 @@ func (n *node) channel(id channel.ID) *paymentChannel {
 
 func (n *node) HandleProposal(req *client.ChannelProposal, res *client.ProposalResponder) {
+	if len(req.PeerAddrs) != 2 {
+		log.Fatal("Channels with more than 2 participants are currently not supported")
```
```suggestion
		log.Fatal("Only channels with two participants are currently supported")
```

matthiasgeihs replied:
done

ggwpez commented on cmd/demo/node_setup.go:
```diff
@@ -40,8 +39,8 @@ var (
-	rng := rand.New(rand.NewSource(0x280a0f350eec))
-	appDef := wtest.NewRandomAddress(rng)
+
+	appDef := ewallet.AsWalletAddr(common.Address{}) // dummy app def
```
```suggestion
	appDef := &ewallet.Address{} // dummy app def
``` 
or just remove the line and do `payment.SetAppDef(&ewallet.Address{})`.

matthiasgeihs replied:
done

ggwpez reviewed (changes requested):
Please just squash the commits, looks good otherwise.

ggwpez commented:
I will quickly rebase the branch and then merge.
